src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -6083,7 +6141,7 @@ private static bool DoDnsCallback(IAsyncResult result, MultipleAddressConnectAsy
             return DoMultipleAddressConnectCallback(PostOneBeginConnect(context), context);
         }
 
-        private class MultipleAddressConnectAsyncResult : ContextAwareResult
+        private sealed class MultipleAddressConnectAsyncResult : ContextAwareResult
```

Comment:
> What caused the reversion? Was there code outside of NetworkInformation that was inheriting from the type?

`SafeCloseSocket` is extended by `SafeCloseSocketAndEvent`. There were a few other changes that will bubble up to System.Net.Primitives. These were expected and luckily not that complicated given that we've started working on PAL before we completely understand how these types are layered.

> It's a private type that AFAICT is not an extension point.
> For any internal-only types, in general I believe we should have a seal-by-default stance
> Sealing has potential performance benefits as well as potential usability benefits around understanding the extensibility potential of a type.

Makes sense. Let's add it to the coding style rules and make the code uniform (if we change this type, let's change others in a similar way).